docs/extensions/project-roles.md

```diff
@@ -0,0 +1,39 @@
+# Extending project roles
+
+The `Project` resource allows to specify a list of roles for every member (`.spec.members[*].roles`).
+There are a few standard roles defined by Gardener itself:
+
+* `owner` (describes the owner/main contact of the project (as of today only one owner can be specified))
+* `admin` (describes administrators with full read/write access to all resources concerning the project)
+* `viewer` (describes members with limited read access to some resources concerning the project)
+
+However, extension controllers running in the garden cluster may also create `CustomResourceDefinition`s that project members might be able to CRUD.
+For this purpose Gardener also allows to specify extension roles.
+
+An extension role is prefixed with `extension:`, e.g.
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: Project
+metadata:
+  name: dev
+spec:
+  members:
+  - apiGroup: rbac.authorization.k8s.io
+    kind: User
+    name: alice.doe@example.com
+    roles:
+    - "admin"
+    - "owner"
+    - "extension:foo"
```

Comment:
Currently if I update the project spec from the example above to have the role `"extension:bar"` instead of the role `"extension:foo"` the ClusterRole with `gardener.cloud:extension:project:dev:foo`won't be deleted. Shouldn't we delete stale ClusterRoles which are no longer part of the roles.

Reply:
Why are you saying this? I cannot reproduce it. Can you provide steps? What I did:
1. Add `extension:foo` role to a user
1. `k get clusterrole | grep exten` shows `gardener.cloud:extension:project:dev:foo` only
1. Update `extension:foo` to `extension:bar`
1. `k get clusterrole | grep exten` shows `gardener.cloud:extension:project:dev:bar` only

Reply:
Sorry, my mistake. You are right.